src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -142,18 +142,29 @@ void LiveFunctionsDataView::UpdateSelectedFunctionId() {
   selected_function_id_ = app_->GetHighlightedFunctionId();
 }
 
-std::vector<uint64_t> LiveFunctionsDataView::GetFunctionTimerDurations(uint64_t function_id) {
+void LiveFunctionsDataView::UpdateTimerDurations() {
```

Comment:
This is doing more work than before. Do we think there might be performance consequences?

Reply:
Actually, it does less. Previously for each `GetFunctionTimerDurations` we used to go though all the timers, select only those that have the correct function_id, put it in a vector (that's what `TimeGraph::GetAllTimersForHookedFunction` does), then we used to get the timer durations, sort them and put the in the map. And this had to be repeated for each function from the very beginning. 

Now we go though all the timers once and group the timer durations for all the distinct scope_id.

Reply:
I think this function should be in `ThreadTrackDataProvider` and the calculation should be invoked from `ThreadTrackDataProvider::OnCaptureComplete`. `ThreadTrackDataProvider` should expose the ready-made hash map via a getter and `LiveFunctionsDataView` should get access via a pointer to `ThreadTrackDataProvider` in its constructor.

Reply:
Done. Almost. Passing the pointer in `LiveFunctionsDataView` doesn't work, as `LiveFunctionsDataView` stays alive for the whole run of the client. But I make calls to the new method `ThreadTrackDataProvider::GetSortedTimerDurationsForScopeId` that queries the map.